### ISSUE TITLE
Class methods are no longer picked up in hidden.rbi

### ISSUE THREAD

**Issue by connorshea** [sorbet/sorbet#1832]
I noticed that Sorbet has stopped including class methods (`self.method_name`) in `hidden.rbi` after upgrading a few days ago.

`hidden.rbi` on `0.4.4755`: https://github.com/connorshea/VideoGameList/blob/06f3e6fb35cbbf81eac891022d6c09946368bdb1/sorbet/rbi/hidden-definitions/hidden.rbi

`hidden.rbi` on `0.4.4770`: https://github.com/connorshea/VideoGameList/blob/968bd637609bd43970c787e410f5683bbcbcedc7/sorbet/rbi/hidden-definitions/hidden.rbi

You'll notice that the `hidden.rbi` on 4770 is a few thousand lines shorter than the `hidden.rbi` on 4755.

You can also reproduce this much more quickly/more simply with a basic project where the only gems are sorbet-related:

```ruby
# frozen_string_literal: true

source "https://rubygems.org"

git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }

gem 'sorbet'
gem 'sorbet-runtime'
gem 'sorbet-static'
```

I've also tested it on 4761 and have the same issue, so it's something that changed in `x > 4755 && x <= 4761`.

This makes Sorbet fail typechecking on code that was previously perfectly valid.

OS: macOS Mojave 10.14.6
Ruby: 2.6.3
Sorbet: 0.4.4761 - 0.4.4782

**Comment by jasnow** [comment]
@connorshea - Good catch

**Comment by jaredbeck** [comment]
> I've also tested it on 4761 and have the same issue, so it's something that changed in x > 4755 && x <= 4761.

According to my testing, this issue was introduced in `0.4.4761`, making`0.4.4756` the highest version that does not have this issue.

**Comment by jaredbeck** [comment]
### Example of disappearing class method

```
# app/models/redacted.rb
class Redacted < ApplicationRecord
  def self.redacted(p)

# hidden.rbi in 0.4.4756
class Redacted
  def self.redacted(p); end
end

# hidden.rbi in 0.4.4761 lacks the above `def`
```

**Comment by connorshea** [comment]
I used `bundle open sorbet` to edit the gem and confirmed that commenting out this line fixes the problem:

https://github.com/sorbet/sorbet/commit/9c9b6ec0623c114809b388df404ea337c19e58c8#diff-d5f7a3939d4f661ffe6736aa20031b79R195

cc: @gdritter-stripe

**Comment by jaredbeck** [comment]
Confirming: Seems fixed in #1868 (released in 0.4.4811). Thanks @gdritter-stripe
